oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/index/ElasticsearchIndexEditorProvider.java

```diff
@@ -49,15 +48,15 @@ public ElasticsearchIndexEditorProvider(@NotNull ElasticsearchIndexCoordinateFac
                                            @NotNull NodeBuilder definition, @NotNull NodeState root,
                                            @NotNull IndexUpdateCallback callback) throws CommitFailedException {
         if (TYPE_ELASTICSEARCH.equals(type)) {
-            checkArgument(callback instanceof ContextAwareCallback, "callback instance not of type " +
-                    "ContextAwareCallback [%s]", callback);
-            IndexingContext indexingContext = ((ContextAwareCallback)callback).getIndexingContext();
+            assert callback instanceof ContextAwareCallback : "callback instance not of type " +
```

Comment:
I am not very sure about this - but I think assertions disabled by default (like you need to pass -ea in the java args to enable them).